python/mirascope/llm/formatting/types.py

```diff
@@ -88,79 +127,43 @@ def recommend_book(genre: str) -> list[llm.Message]:
           ]
 
       response: llm.Response[Book] = recommend_book("fantasy")
+      print(response.format_info)
+
       book: Book = response.format()
       print(f"{book.title} by {book.author}")
     ```
     """
 
-    name: str
-    """The name of the response format."""
-
-    description: str | None
-    """A description of the response format, if available."""
-
-    schema: dict[str, object]
-    """JSON schema representation of the structured output format."""
+    spec: FormatSpec
+    """The detailed specification of the format, including name, schema, etc."""
 
     mode: FormattingMode
     """The mode of the response format. 
     
     Determines how the LLM call may be modified in order to extract the expected format.
+
+    If this mode was not provided by the decorator (via the `FormatSpec`), then it will
+    be chosen by the llm client being used.
     """
 
     format: type[BaseModel]
     """The associated format that this `FormatInfo` describes."""
 
-
-ConcreteFormattingMode = Literal[
-    "strict",
-    "tool",
-    "json",
-]
-"""Concrete mode for formatting responses with a particular LLM.
-
-The semantics for modes "strict", "tool", and "json" are as described in `FormattingMode`. 
-
-"strict-or-tool" and "strict-or-json" are not present because they will have resolved to one of "strict", "tool", or "json".
-"""
-
-
-@dataclass(kw_only=True)
-class ResolvedFormatInfo:
-    """Resolved format info for direct processing by LLM.
-
-    A `FormatInfo` is converted into a `ResolvedFormatInfo` using
-    `ResolvedFormatInfo.from_format_info(...)`. Based on information about the model's capabilities,
-    it will include a concrete formatting mode. It may also contain formatting instructions
-    that will be added to the model's system message. Those formatting instructions will be
-    the `FormatInfo`'s system instructions if present, or may be auto-generated instructions
-    (e.g. a dump of the expected JSON schema in JSON mode) if no custom formatting instructions
-    were provided.
-    """
-
-    # TODO: Make ResolvedFormatInfo accessible from the response somehow, which will make
-    # it possible for users of e.g. "strict-or-tool" to see which concrete formatting mode
-    # was invoked, as well as making formatting_instructions more discoverable.
-
-    mode: ConcreteFormattingMode
-    """The concrete formatting mode that will be used."""
-
-    info: FormatInfo
-    """The FormatInfo this ResolvedFormatInfo was resolved from."""
-
     formatting_instructions: str | None
```

Comment:
Note: Formatting instructions need to be on the `FormatInfo`, not the `FormatSpec`, because at decorator time (when we create the spec), we may not have a mode, and the autogenerated instructions are mode-specific.

Reply:
We keep the `None` case because strict mode doesn't require formatting instructions.